File: Adapter/cpp-source/TextJsonAdapter.cpp
//
// Created by Pavel Akhtyamov on 26/03/2018.
//

#include "TextJsonAdapter.h"

std::unordered_map<std::string, std::string> TextJsonAdapter::ConvertToJson() {
  std::unordered_map<std::string, std::string> result;

  for (auto key_value_pair : GetPairs()) {
    result[key_value_pair.key] = key_value_pair.value;
  }
  return result;
}


void TextJsonAdapter::Write(const std::string &filename) {
  std::unordered_map<std::string, std::string> pairs = ConvertToJson();
  storage_.SetData(pairs);
  storage_.WriteJson(filename);
}

File: Adapter/python-source/TextJsonAdapter.py
from JsonStorage import JsonStorage
from TextStorage import TextStorage


class TextJsonAdapter(TextStorage):
    def __init__(self):
        super().__init__()
        self.json_storage = JsonStorage()

    def write(self, filename):
        pairs = self.convert_to_json()
        self.json_storage.data = pairs
        self.json_storage.write_json(filename)

    def convert_to_json(self):
        pairs = {}
        for pair in self.pairs:
            pairs[pair.key] = pair.value
        return pairs
